app/src/main/java/com/pvp/app/ui/screen/profile/ProfileViewModel.kt

```diff
@@ -42,6 +51,29 @@ class ProfileViewModel @Inject constructor(
             )
         )
 
+    suspend fun deleteAccount(): Result<Boolean> {
```

Comment:
We are pretty much handling the whole logic here, which I tend to think shouldn't be done in the ViewModel. ViewModel should act as Controller would in MVC architecture - controlling what is the logic of process flow pretty much, what is called and on what conditions.

I believe that user removing logic should be handled in the AuthenticationService.